homeassistant/components/oralb/sensor.py

```diff
@@ -1,35 +1,66 @@
 """Platform for sensor integration."""
 from __future__ import annotations
 
+from dataclasses import dataclass
 from datetime import timedelta
 import logging
 
-from oralb import OralB
-import voluptuous as vol
-
 from homeassistant import config_entries
-from homeassistant.components import bluetooth
 from homeassistant.components.sensor import (
-    PLATFORM_SCHEMA,
     SensorDeviceClass,
     SensorEntity,
+    SensorEntityDescription,
     SensorStateClass,
 )
-from homeassistant.const import CONF_MAC, CONF_NAME, PERCENTAGE, TIME_SECONDS
+from homeassistant.const import CONF_MAC, PERCENTAGE, TIME_SECONDS
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import ConfigEntryNotReady
-import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
-SCAN_INTERVAL = timedelta(seconds=3)
-
-# Validation of the user's configuration
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_NAME): cv.string,
-        vol.Required(CONF_MAC): cv.string,
-    }
+SCAN_INTERVAL = timedelta(seconds=1)
+
+
+@dataclass
+class OralBSensorEntityDescription(SensorEntityDescription):
+    """Provide a description of a OralB sensor."""
+
+    # For backwards compat, allow description to override unique ID key to use
+    unique_id: str | None = None
+
+
+SENSORS = (
+    OralBSensorEntityDescription(
+        key="battery",
+        name="OralB Battery",
+        unique_id="oralb_battery",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    OralBSensorEntityDescription(
+        key="status",
+        name="OralB Status",
+        unique_id="oralb_status",
+        icon="mdi:toothbrush-electric",
+    ),
+    OralBSensorEntityDescription(
+        key="brush_time",
+        name="OralB Brush Time",
```

Comment:
No need to prefix the names with OralB. Since you use entity names, the device name is automatically prefixed. 
```suggestion
        name="Brush Time",
```